sycl/source/detail/device_info.hpp

```diff
@@ -740,186 +740,189 @@ struct get_device_info_impl<
-           matrix_type::fp32, matrix_type::fp16},
-          {0, 0, 0, 8, 32, 16, matrix_type::fp16, matrix_type::fp16,
-           matrix_type::fp32, matrix_type::fp16},
-          {0, 0, 0, 32, 8, 16, matrix_type::fp16, matrix_type::fp16,
-           matrix_type::fp32, matrix_type::fp16},
-          {0, 0, 0, 16, 16, 16, matrix_type::fp16, matrix_type::fp16,
-           matrix_type::fp16, matrix_type::fp32},
-          {0, 0, 0, 8, 32, 16, matrix_type::fp16, matrix_type::fp16,
-           matrix_type::fp16, matrix_type::fp32},
-          {0, 0, 0, 32, 8, 16, matrix_type::fp16, matrix_type::fp16,
-           matrix_type::fp16, matrix_type::fp32}};
-      std::vector<combination> sm_72_combinations = {
-          {0, 0, 0, 16, 16, 16, matrix_type::sint8, matrix_type::sint8,
-           matrix_type::sint32, matrix_type::sint32},
-          {0, 0, 0, 8, 32, 16, matrix_type::sint8, matrix_type::sint8,
-           matrix_type::sint32, matrix_type::sint32},
-          {0, 0, 0, 32, 8, 16, matrix_type::sint8, matrix_type::sint8,
-           matrix_type::sint32, matrix_type::sint32},
-          {0, 0, 0, 16, 16, 16, matrix_type::uint8, matrix_type::uint8,
-           matrix_type::sint32, matrix_type::sint32},
-          {0, 0, 0, 8, 32, 16, matrix_type::uint8, matrix_type::uint8,
-           matrix_type::sint32, matrix_type::sint32},
-          {0, 0, 0, 32, 8, 16, matrix_type::uint8, matrix_type::uint8,
-           matrix_type::sint32, matrix_type::sint32}};
-      std::vector<combination> sm_80_combinations = {
-          {0, 0, 0, 16, 16, 8, matrix_type::tf32, matrix_type::tf32,
-           matrix_type::fp32, matrix_type::fp32},
-          {0, 0, 0, 16, 16, 16, matrix_type::bf16, matrix_type::bf16,
-           matrix_type::fp32, matrix_type::fp32},
-          {0, 0, 0, 8, 32, 16, matrix_type::bf16, matrix_type::bf16,
-           matrix_type::fp32, matrix_type::fp32},
-          {0, 0, 0, 32, 8, 16, matrix_type::bf16, matrix_type::bf16,
-           matrix_type::fp32, matrix_type::fp32},
-          {0, 0, 0, 8, 8, 4, matrix_type::fp64, matrix_type::fp64,
-           matrix_type::fp64, matrix_type::fp64}};
-      if (ComputeCapability >= 8.0) {
-        sm_80_combinations.insert(sm_80_combinations.end(),
-                                  sm_72_combinations.begin(),
-                                  sm_72_combinations.end());
-        sm_80_combinations.insert(sm_80_combinations.end(),
-                                  sm_70_combinations.begin(),
-                                  sm_70_combinations.end());
-        return sm_80_combinations;
-      } else if (ComputeCapability >= 7.2) {
-        sm_72_combinations.insert(sm_72_combinations.end(),
-                                  sm_70_combinations.begin(),
-                                  sm_70_combinations.end());
-        return sm_72_combinations;
-      } else if (ComputeCapability >= 7.0)
-        return sm_70_combinations;
+    try {
```

Comment:
What about [function `try...catch`](https://en.cppreference.com/w/cpp/language/function-try-block) block? I think it would be more readable by allowing to reduce indentation level

Reply:
Cool, thank you!

Reply:
I feel like this change actually harms readability in this case. While reducing the indentation level is nice, I think the intent would be much clearer if the try block was limited to the only callee we expect the exception to be thrown from rather than the entire (and rather large) function body.

Reply:
It's impossible to do that.
```
architecture DeviceArch = get_device_info_impl<
        ext::oneapi::experimental::architecture,
        ext::oneapi::experimental::info::device::architecture>::get(Dev);
```
may throw the exception - but DeviceArch is used everywhere below in the func, and there is no valid initial value for the variable of `architecture` type.

Reply:
How about moving this call and the exception handling to a small helper that would return something like `std::optional<architecture>`?

Reply:
Sounds good to me, applied in https://github.com/intel/llvm/pull/12866/commits/9a14c220a354e4219a83c9f588463b3ee62e3f62